packages/manager/apps/web/client/app/hosting/multisite/hosting-multisite.controller.js

```diff
@@ -129,6 +135,77 @@ angular
               }
             });
           })
+          .then(async () => {
+            $scope.user = await WucUser.getUser();
+          })
+          .then(() => {
+            const getNichandleAttached = async (
+              diagnosticSuffix,
+              domain,
+              nichandle,
+            ) => {
+              const serviceInfos = await HostingDomain.getServiceInfosForZone(
+                domain.name,
+              );
+              const { contactTech, contactAdmin } = serviceInfos;
+
+              const isDnsAttachedToNic =
+                nichandle === contactAdmin || nichandle === contactTech;
+              return {
+                diagnosticState: diagnosticSuffix[domain.name],
+                isDnsAttachedToNic,
+              };
+            };
+
+            const promises = $scope.domains.list.results.map(async (domain) => {
```

Comment:
**nitpick:** `async ... await` does not couple well with the AngularJS digest cycle. please use `$q`.